File: Data Structures & Algorithms/string-encode-and-decode/submission-4.js
class Solution {
    /**
     * @param {string[]} strs
     * @returns {string}
     */
    encode(strs) {
        
       let res = "";
        for(let s of strs){
            res += s.length + "#" + s;
        }

        return res;
    }

    /**
     * @param {string} str
     * @returns {string[]}
     */
    decode(str) {

        let l = 0;
        const res = [];
        for (let r = 0; r<str.length; r++){
            if(str[r] != "#")
                continue;

            let num = parseInt(str.substring(l,r));
            res.push(str.substring(r+1, r+num+1));

            l = r+num+1;
            r += num;
        }

        return res;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-4.cs
public class Solution {
    public int Trap(int[] height) {
        int l = 0;
        int r = height.Length - 1;

        int lm = height[l];
        int rm = height[r];

        int res = 0;
        while (l < r) {
            if (lm < rm) {
                l++;
                if (lm < height[l])
                    lm = height[l];
                else
                    res += lm - height[l];

            } else {
                r--;
                if (rm < height[r])
                    rm = height[r];
                else
                    res += rm - height[r];
            }
        }

        return res;
    }
}
